packages/tools/client-debugger/client-debugger-view/src/Debugger.tsx

```diff
@@ -40,82 +54,98 @@ export interface FluidClientDebuggersProps {
  * @remarks If no debugger has been initialized, will display a note to the user and a refresh button to search again.
  */
 export function FluidClientDebuggers(props: FluidClientDebuggersProps): React.ReactElement {
-	const debuggerRegistry: DebuggerRegistry = getDebuggerRegistry();
+	const [containers, setContainers] = React.useState<ContainerMetadata[]>([]);
+	const [menuSelection, setMenuSelection] = React.useState<string>("");
```

Comment:
I wonder if it would be worth using a discriminated union here instead of a string. That way we can store any other state associated with the selection (e.g. containerId for the "containers" tab) alongside it.

Otherwise, we probably need to reset the containerId selection any time we change from "containers" to a different selection, etc..

Reply:
True, I like the discriminated union approach, I'll move to that.